GoogleForCreators/web-stories-wp
# Animation: Background Zoom

## Summary
Adds foundation for background animation effects & adds in background zoom effect.

Behavior spec for zoom scale slider:
- When added should initialize to same position as bg scale slider
- Should mimic the exisiting bg scale slider, but the static bg scale slider should be the end scale, and the bg zoom scale slider should be the start scale

## Relevant Technical Choices
- Adds ability to target leaf node for animation since these bg animations aren't possible from the wrapper component
- Normalized Range slider interpreted at effect level with story element info
- Foundation for background effects in animation dir
- Hoisted up common range calculations into animations dir
- Adds animation default behavior based off of element properties in `getPanels()` function

## To-do
Didn't do this off of any existing designs, so need to get design approval/feedback.

Will also have a separate PR that adds a background pan effect.

## User-facing changes
None, all behind feature flag. 

But if you enable the relevant animation experiments you should now be able to add Zoom animations to background elements.
<img width="352" alt="Screen Shot 2020-11-13 at 11 20 19 AM" src="https://user-images.githubusercontent.com/35983235/99109599-4fe2e200-25a6-11eb-99d7-a52843234a77.png">
<img width="304" alt="Screen Shot 2020-11-13 at 11 20 03 AM" src="https://user-images.githubusercontent.com/35983235/99109600-507b7880-25a6-11eb-936f-066b0d6b6900.png">


## Testing Instructions
Enable all animation experiments -> Add a background to your story -> Apply a zoom animation to that background -> play with different variations of:
Static end scale:
<img width="502" alt="Screen Shot 2020-11-13 at 11 55 09 AM" src="https://user-images.githubusercontent.com/35983235/99110116-21b1d200-25a7-11eb-9512-78b9c3a72b5a.png">

Zoom anim scale from:
<img width="301" alt="Screen Shot 2020-11-13 at 11 55 22 AM" src="https://user-images.githubusercontent.com/35983235/99110185-3aba8300-25a7-11eb-8ae2-5ecb48513b41.png">

Make sure to test on all bg file types both in the editor and the story preview:
- GIF
- Video
- Image

---



Fixes #3529
